<documentation title="Parameter Spacing in Function Calls">
    <standard>
    <![CDATA[
     There must be one space between a comma and a parameter in a function call.
    ]]>
    </standard>
    <code_comparison>
        <code title="Valid: spaces between parameters">
        <![CDATA[
$var = foo($bar<em>, </em>$baz<em>, </em>$quux);
        ]]>
        </code>
        <code title="Invalid: no space between commas and parameters">
        <![CDATA[
$var = foo($bar<em>,</em>$baz<em>,</em>$quux);
        ]]>
        </code>
    </code_comparison>
</documentation>
