From e5722d0358e94e3c7511df46a1eada21a021b894 Mon Sep 17 00:00:00 2001
From: Tom Kirkpatrick <tom@systemseed.com>
Date: Tue, 15 Nov 2011 09:34:16 +0100
Subject: [PATCH] Fix for multipage scroll issue

---
 multipage/multipage.js |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/multipage/multipage.js b/multipage/multipage.js
index 5e10691..02d53a9 100755
--- a/multipage/multipage.js
+++ b/multipage/multipage.js
@@ -140,6 +140,7 @@ Drupal.multipageControl.prototype = {
    */
   nextPage: function () {
     this.wrapper.next().data('multipageControl').focus();
+    $('html, body').scrollTop($("#page").offset().top);
   },
   
   /**
@@ -147,6 +148,7 @@ Drupal.multipageControl.prototype = {
    */
   previousPage: function () {
     this.wrapper.prev().data('multipageControl').focus();
+    $('html, body').scrollTop($("#page").offset().top);
   },
 
   /**
-- 
1.7.4.4

