Index: sites/all/modules/field_group/field_group.js
===================================================================
--- sites/all/modules/field_group/field_group.js	(revision 17762)
+++ sites/all/modules/field_group/field_group.js	(working copy)
@@ -137,6 +137,7 @@
           }
           wrapper.animating = false;
         }
+        $wrapper.toggleClass('collapsed');
         return false;
       });
       

