diff --git a/multipage/multipage.js b/multipage/multipage.js
index b252b20..e4e21b3 100755
--- a/multipage/multipage.js
+++ b/multipage/multipage.js
@@ -209,12 +209,12 @@ Drupal.theme.prototype.multipage = function (settings) {
   var controls = {};
   controls.item = $('<span class="multipage-button"></span>');
   controls.item.append(controls.nextLink = $('<input type="button" class="form-submit multipage-link-next" value="" />').val(controls.nextTitle = Drupal.t('Next page')));
-  controls.item.append(controls.previousLink = $('<a class="multipage-link-previous" href="#"></a>'));
+  controls.item.append(controls.previousLink = $('<input type="button" class="form-submit multipage-link-previous" value="" />').val(controls.nextTitle = Drupal.t('Previous')));
   if (!settings.has_next) {
     controls.nextLink.hide();
   }
-  if (settings.has_previous) {
-    controls.previousLink.append(controls.previousTitle = $('<strong></strong>').text(Drupal.t('Previous')));
+  if (!settings.has_previous) {
+    controls.previousLink.hide();
   }
   return controls;
 };
