diff --git a/field_group.module b/field_group.module
index fc6aaba..8573b1f 100755
--- a/field_group.module
+++ b/field_group.module
@@ -278,6 +278,13 @@ function field_group_field_group_formatter_info() {
         'instance_settings' => array('description' => '', 'show_label' => 1, 'label_element' => 'h3', 'effect' => 'none', 'speed' => 'fast', 'classes' => '', 'required_fields' => 1, 'id' => ''),
         'default_formatter' => 'open',
       ),
+      'html5' => array(
+        'label' => t('HTML5'),
+        'description' => t('This fieldgroup renders the inner content in a semantic HTML5 wrapper'),
+        'format_types' => array('section', 'article', 'header', 'footer', 'aside'),
+        'instance_settings' => array('semantic_wrapper' => '', 'classes' => ''),
+        'default_formatter' => 'section',
+      ),
       'fieldset' => array(
         'label' => t('Fieldset'),
         'description' => t('This fieldgroup renders the inner content in a fieldset with the titel as legend.'),
@@ -330,6 +337,13 @@ function field_group_field_group_formatter_info() {
         'instance_settings' => array('description' => '', 'show_label' => 1, 'label_element' => 'h3', 'effect' => 'none', 'speed' => 'fast', 'classes' => ''),
         'default_formatter' => 'collapsible',
       ),
+      'html5' => array(
+        'label' => t('HTML5'),
+        'description' => t('This fieldgroup renders the inner content in a semantic html5 wrapper'),
+        'format_types' => array('section', 'article', 'header', 'footer', 'aside'),
+        'instance_settings' => array('semantic_wrapper' => '', 'classes' => ''),
+        'default_formatter' => 'section',
+      ),
       'fieldset' => array(
         'label' => t('Fieldset'),
         'description' => t('This fieldgroup renders the inner content in a fieldset with the titel as legend.'),
@@ -693,6 +707,22 @@ function field_group_pre_render_div(&$element, $group, &$form) {

 /**
  * Implements field_group_pre_render_<format-type>.
+ * Format type: HTML5.
+ *
+ * @param $element The field group form element.
+ * @param $group The Field group object prepared for pre_render.
+ * @param $form The root element or form.
+ */
+function field_group_pre_render_html5(&$element, $group, &$form) {
+  $element += array(
+    '#type' => 'markup',
+    '#prefix' => '<' . $group->format_settings['formatter'] . ' id="' . $element['#id'] . '" class="field-group-format ' . $group->classes . '">',
+    '#suffix' => '</' . $group->format_settings['formatter'] . '>',
+  );
+}
+
+/**
+ * Implements field_group_pre_render_<format-type>.
  * Format type: Accordion.
  *
  * @param $element The field group form element.
