diff --git a/field_group.install b/field_group.install
index 932e260..9e40b38 100644
--- a/field_group.install
+++ b/field_group.install
@@ -163,6 +163,7 @@ function field_group_install() {
         'formatter' => preg_match("/fieldset/", $group->settings['form']['style']) ? 'collapsible' : 'collapsed',
         'instance_settings' => array(),
       );
+      $new->identifier = $new->group_name . '|' . $new->entity_type . '|' . $new->bundle . '|' . $new->mode;
       ctools_export_crud_save('field_group', $new);
 
       // The full node.
@@ -174,6 +175,7 @@ function field_group_install() {
         'formatter' => 'collapsible',
         'instance_settings' => array(),
       );
+      $new->identifier = $new->group_name . '|' . $new->entity_type . '|' . $new->bundle . '|' . $new->mode;
       ctools_export_crud_save('field_group', $new);
 
       // The teaser node.
@@ -185,6 +187,7 @@ function field_group_install() {
         'formatter' => 'collapsible',
         'instance_settings' => array(),
       );
+      $new->identifier = $new->group_name . '|' . $new->entity_type . '|' . $new->bundle . '|' . $new->mode;
       ctools_export_crud_save('field_group', $new);
 
     }
